tests/Donation/TransactionSubscriberTest.php

```diff
@@ -0,0 +1,109 @@
+
+namespace Tests\AppBundle\Donation;
+
+use AppBundle\DataFixtures\ORM\LoadDonationData;
+use AppBundle\Repository\DonationRepository;
+use AppBundle\Repository\TransactionRepository;
+use Tests\AppBundle\Controller\ControllerTestTrait;
+use Tests\AppBundle\MysqlWebTestCase;
+
+/**
+ * @group functional
+ */
+class TransactionSubscriberTest extends MysqlWebTestCase
+{
+    use ControllerTestTrait;
+    /**
+     * @var DonationRepository
+     */
+    private $donationRepository;
+    /**
+     * @var TransactionRepository
+     */
+    private $transactionRepository;
+
+    public function payloadProvider(): iterable
+    {
+        yield 'transaction1' => ['424241'];
+        yield 'transaction2' => ['424242'];
+        yield 'transaction3' => ['424243'];
+    }
+
+    /**
+     * @dataProvider payloadProvider
+     */
+    public function testOnPayboxIpnResponse(string $transactionId): void
+    {
+        $params = $this->createPayload($transactionId);
+        static::assertNull($this->transactionRepository->findByPayboxTransactionId($transactionId));
+
+        $this->createSignature($params);
+
+        $this->client->request('POST', '/don/payment-ipn/1528114726', $params);
+
+        static::assertSame('OK', $this->client->getResponse()->getContent());
+
+        static::assertNotNull($this->transactionRepository->findByPayboxTransactionId($transactionId));
+    }
+
+    /**
+     * @depends testOnPayboxIpnResponse
+     */
+    public function testAllTransactionCreated(): void
+    {
+        $transactions = $this->transactionRepository->findAllSuccessfulTransactionByEmail('jacques.picard@en-marche.fr');
+
+        // 4 because there are the initiale transaction when donationFixtures are loaded
+        static::assertCount(4, $transactions);
+    }
+
+    private function createSignature(array &$params): void
+    {
+        $queryParams = http_build_query($params);
+        $privateKey = openssl_pkey_get_private($this->container->getParameter('env(SSL_PRIVATE_KEY)'));
+        openssl_sign($queryParams, $signature, $privateKey, 'sha1WithRSAEncryption');
+        $params['Sign'] = base64_encode($signature);
+    }
+
+    private function createPayload(string $transactionId): array
+    {
+        $donation = $this->donationRepository->findAllSubscribedDonationByEmail('jacques.picard@en-marche.fr')[0];
+
+        return [
+            'id' => $donation->getUuid()->toString().'_test',
+            'authorization' => 'XXXXXX',
+            'result' => '00000',
+            'transaction' => $transactionId,
+            'amount' => '50000',
+            'date' => '01062018',
+            'time' => '14:52:22',
+            'card_type' => 'MasterCard',
+            'card_end' => '1812',
+            'card_print' => '5B434C778490889697170E225029F56AFF19CA47',
+            'subscription' => '1234',
+        ];
+    }
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->init([
+            LoadDonationData::class,
```

Comment:
Master of all lords (tmtc @B-Galati) teached me we should declare fixtures dependencies explicitly to avoid confusion.

Please add `LoadAdherentData::class` since `LoadDonationdata::class` depends on it.

Reply:
I think it's useless because if tomorrow we decide to remove LoadAdherent, the fixture will be loaded for nothing and technically the test does not use Adherents datas.

Btw Adherent's data are used to generate donations data by using data already set in adherent but we can totaly remove it and set data with a faker or ... , the only link between donation and user account is the email address . there are no foreign key, ... .
```php
        $donation = new Donation(
            $uuid = Uuid::uuid4(),
            $uuid->toString().'_'.$this->slugify->slugify($adherent->getFullName()),
            $amount * 100,
            $adherent->getGender(),
            $adherent->getFirstName(),
            $adherent->getLastName(),
            $adherent->getEmailAddress(),
            $adherent->getPostAddress(),
            $adherent->getPhone(),
            '127.0.0.1',
            $duration
        );
```

Reply:
fair enough !